### PR TITLE
Add package version in config 75

### PR DESCRIPTION
**Pull request by geauxdojang**

### CODE REVIEW

**Comment by camorosi** [comment]
Hi, this is a good start. I have some suggestions:

1. `update_package_versions_in_config_yaml` is no longer used, suggest removing
2. Instead of having the pins version checked in dpinput_map, it would be better to check within dpconf_get, which is typically called first in a normal project workflow. Alternatively, you could create a util function such as 'check_pins_compatability` that's called in both places.
3. I think the logic behind the pin version check is not quite right: I think the best check is to make sure all three of these things are true: 1) pins_version key is in the config file, 2) read_conf_file$pins_version >= "1.2.0", 3) pins packageVersion >= "1.2.0". If any of these three are not true, then alert with the error message about downgrading packages. 
4. The message about downgrading packages should also include a line about installing legacy pins `remotes::install_github(repo = "amashadihossein/pins")`

**Comment by camorosi** [comment]
Looks good to me; ready for review. 

Changes are: 
* `dp_init` adds pin_version key to daap_config.yaml file (will be v 1.2.0 or greater for all new data products)
*  `dpconf_get` and `dpinput_map` now call helper function `check_pins_compatibility` to make sure than the correct version of pins is used for new data products

File: R/dp_init.R [hunk 1]
```diff
@@ -168,6 +168,10 @@ dp_init <- function(project_path = fs::path_wd(),
     fs::dir_create(glue::glue("{project_path}/output_files"))
   }
 
+  pins_version <- utils::packageVersion("pins")
+
+  is_legacy <- pins_version >= '1.2.0'
```
Comment (camorosi) [inline R/dp_init.R:173]: @geauxdojang this flag is backwards: we consider data products legacy if they use the old version of pins

File: R/utils.R [hunk 2]
```diff
@@ -288,3 +288,35 @@ flname_xos_get <- function(fl, package = "dpbuild") {
...
+
+#' @title Check pins package compatibility
+#' @description Check pins package compatibility
+#' @param is_legacy A boolean variable to indicate if the pins version is legacy one
+#' @keywords internal
+check_pins_compatibility <- function(){
+  read_conf_file <- dpbuild:::dpconf_read(project_path = ".")
+
+  is_legacy_key_in_config <- "is_legacy" %in% names(read_conf_file)
+  is_legacy_pins_in_config <- as.logical(read_conf_file$is_legacy)
```
Comment (camorosi) [inline R/utils.R:301]: For simplicity, I think you should combine these two to one single option: "is_legacy_dp", which would only be true if both of these conditions are met. Note that if the key doesn't exist, the second boolean here will actually be NA, not FALSE, so it won't behave as expected in the all statement below. Also, I think the as.logical is unnecessary here

File: R/dp_init.R [hunk 3]
```diff
@@ -168,6 +168,10 @@ dp_init <- function(project_path = fs::path_wd(),
     fs::dir_create(glue::glue("{project_path}/output_files"))
   }
 
+  pins_version <- utils::packageVersion("pins")
+
+  is_legacy <- pins_version < '1.2.0'
```
Comment (amashadihossein) [inline R/dp_init.R:173]: I thought this was a syntax error! Cool that it works!

**Review by amashadihossein — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**